cypress/integration/settings/circulation/fixed-due-date-schedules/schedule-does-not-cover-the-test-date.spec.js

```diff
@@ -0,0 +1,177 @@
+                  },
+                })
+                  .then(() => {
+                    cy.getServicePointsApi();
+                    cy.getRequestPolicy({ query: `name=="${REQUEST_POLICY_NAMES.ALLOW_ALL}"` });
+                    cy.getNoticePolicy({ query: `name=="${NOTICE_POLICY_NAMES.SEND_NO_NOTICES}"` });
+                    cy.getOverdueFinePolicy({ query: `name=="${OVERDUE_FINE_POLICY_NAMES.OVERDUE_FINE_POLICY}"` });
+                    cy.getLostItemFeesPolicy({ query: `name=="${LOST_ITEM_FEES_POLICY_NAMES.LOST_ITEM_FEES_POLICY}"` });
+                    cy.getCirculationRules();
+                  })
+                  .then(() => {
+                    rulesDefaultString = Cypress.env(CY_ENV.CIRCULATION_RULES).rulesAsText;
+                    const loanPolicyId = Cypress.env(CY_ENV.LOAN_POLICY).id;
+                    const requestPolicyId = Cypress.env(CY_ENV.REQUEST_POLICY)[0].id;
+                    const noticePolicyId = Cypress.env(CY_ENV.NOTICE_POLICY)[0].id;
+                    const overdueFinePolicyId = Cypress.env(CY_ENV.OVERDUE_FINE_POLICY)[0].id;
+                    const lostItemFeesPolicyId = Cypress.env(CY_ENV.LOST_ITEM_FEES_POLICY)[0].id;
+                    const newRule = `\nm ${Cypress.env(CY_ENV.MATERIAL_TYPES)[0].id}: l ${loanPolicyId} r ${requestPolicyId} n ${noticePolicyId} o ${overdueFinePolicyId} i ${lostItemFeesPolicyId}`;
+
+                    cy.updateCirculationRules({
+                      rulesAsText: rulesDefaultString + newRule,
+                    });
+                  })
+                  .then(() => {
+                    cy.createItemCheckout({
+                      servicePointId: Cypress.env(CY_ENV.SERVICE_POINTS)[0].id,
+                      itemBarcode: ITEM_BARCODE,
+                      userBarcode: USER_BARCODE,
+                      loanDate: moment.utc(),
+                    });
+                  });
+              });
+          });
+      });
+  });
+
+  after(() => {
+    cy.createItemCheckinApi({
+      itemBarcode: ITEM_BARCODE,
+      servicePointId: Cypress.env(CY_ENV.SERVICE_POINTS)[0].id,
+      checkInDate: moment.utc().format(),
+    })
+      .then(() => {
+        cy.deleteUser(userId);
+        cy.getInstance({ limit: 1, expandAll: true, query: `"items.barcode"=="${ITEM_BARCODE}"` })
+          .then((instance) => {
+            cy.deleteItem(instance.items[0].id);
+            cy.deleteHoldingRecord(instance.holdings[0].id);
+            cy.deleteInstanceApi(instance.id);
+          });
+        cy.updateCirculationRules({
+          rulesAsText: rulesDefaultString,
+        });
+        cy.deleteLoanPolicy(Cypress.env(CY_ENV.LOAN_POLICY).id)
+          .then(() => {
+            cy.deleteFixedDueDateSchedule(Cypress.env(CY_ENV.FIXED_DUE_DATE_SCHEDULE).id);
+          });
+      });
+  });
+
+  it('C641: renewing item using a fixed due date loan profile where the fixed due date schedule date range does not cover the test date',
```

Comment:
try to move these points into api steps in related fragements:
"Create a fixed due date schedule that covers the test date
Create a loan policy using that schedule
Create a circulation rule using the loan policy
Check out an item entering a patron and item that meet the circulation rule "

Reply:
I moved some common thing inside methods, but looks like the most of thing are very specific to do fragments from them.